Content.Shared/Weapons/Reflect/ReflectSystem.cs

```diff
@@ -49,72 +49,128 @@ public override void Initialize()
     {
         if (args.Cancelled)
             return;
 
-        if (TryReflectProjectile(uid, uid, args.ProjUid, reflect: component))
-            args.Cancelled = true;
+        if (!TryReflectProjectile(objectUid, objectUid, args.ProjUid))
+            return;
+
+        args.Cancelled = true;
     }
 
-    private bool TryReflectProjectile(EntityUid user, EntityUid reflector, EntityUid projectile, ProjectileComponent? projectileComp = null, ReflectComponent? reflect = null)
+    /// <summary>
+    /// Can a user reflect something that's hit them? Returns true if so, and the best reflector available in the user's equipment.
+    /// </summary>
+    private bool UserCanReflect(EntityUid uid, [NotNullWhen(true)] out EntityUid? bestReflectorUid, EntityUid? projectile = null)
+    {
+        bestReflectorUid = null;
+        ReflectComponent? bestReflector = null;
+        ReflectiveComponent? reflective = null;
+
+        // If it's a projectile trying to hit someone, make sure it can be reflected.
+        if (projectile != null && !TryComp(projectile, out reflective))
+            return false;
+
+        foreach (var entityUid in _inventorySystem.GetHandOrInventoryEntities(uid, SlotFlags.WITHOUT_POCKET))
+        {
+            if (!TryComp<ReflectComponent>(entityUid, out var comp))
+                continue;
+
+            if (!comp.Enabled)
+                continue;
+
+            if (bestReflector != null && bestReflector.ReflectProb >= comp.ReflectProb)
+                continue;
+
+            if (reflective != null && (comp.Reflects & reflective.Reflective) == 0x0)
+                continue;
+
+            bestReflector = comp;
+            bestReflectorUid = entityUid;
+        }
+
+        return bestReflectorUid != null && bestReflector != null;
+    }
+
+    private bool TryReflectProjectile(EntityUid user, EntityUid reflector, EntityUid projectile, ProjectileComponent? projectileComp = null)
     {
-        // Do we have the components needed to try a reflect at all?
         if (
-            !Resolve(reflector, ref reflect, false) ||
+            // Does the reflector actually reflect?
+            !TryComp<ReflectComponent>(reflector, out var reflect) ||
+            // Is it on?
             !reflect.Enabled ||
+            // Is the projectile deflectable?
             !TryComp<ReflectiveComponent>(projectile, out var reflective) ||
+            // Does the deflector deflect the type of projecitle?
             (reflect.Reflects & reflective.Reflective) == 0x0 ||
+            // Is the projectile correctly set up with physics?
             !TryComp<PhysicsComponent>(projectile, out var physics) ||
-            TryComp<StaminaComponent>(reflector, out var staminaComponent) && staminaComponent.Critical ||
+            // If the user of the reflector is a mob with stamina, is it capable of deflecting?
+            TryComp<StaminaComponent>(user, out var staminaComponent) && staminaComponent.Critical ||
```

Comment:
Shouldn't this set the reflector active field to  false? Also this seems like it should be an event subscription

Reply:
You can proc on downed but you can't proc on stam crit. 

Hopefully newmed et al has better event hooks